From 908387692f4427870a78fea52aa032d9367946b4 Mon Sep 17 00:00:00 2001
From: Afzal Mohammed <afzal@ti.com>
Date: Thu, 13 Oct 2011 11:03:46 +0530
Subject: [PATCH 0595/1092] arm:omap:am33xx: IA EVM low level debug

Low level debug support for IA EVM, it uses UART3
(numbering as per TRM). This is referred as UART4
by software, software UART numbering starts from 1.

Signed-off-by: Afzal Mohammed <afzal@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 arch/arm/mach-omap2/include/mach/debug-macro.S | 4 ++++
 arch/arm/plat-omap/include/plat/serial.h       | 1 +
 arch/arm/plat-omap/include/plat/uncompress.h   | 3 +++
 3 files changed, 8 insertions(+)

diff --git a/arch/arm/mach-omap2/include/mach/debug-macro.S b/arch/arm/mach-omap2/include/mach/debug-macro.S
index d7f844a..b9b10e5 100644
--- a/arch/arm/mach-omap2/include/mach/debug-macro.S
+++ b/arch/arm/mach-omap2/include/mach/debug-macro.S
@@ -74,6 +74,8 @@ omap_uart_lsr:	.word	0
 		beq	83f			@ configure UART3
 		cmp	\rp, #AM33XXUART1	@ AM33XX UART offsets different
 		beq	84f			@ configure UART1
+		cmp	\rp, #AM33XXUART4	@ AM33XX UART offsets different
+		beq	85f			@ configure UART1
 		cmp	\rp, #ZOOM_UART		@ only on zoom2/3
 		beq	95f			@ configure ZOOM_UART
 
@@ -105,6 +107,8 @@ omap_uart_lsr:	.word	0
 84:		ldr	\rp, =AM33XX_UART1_BASE
 		and	\rp, \rp, #0x00ffffff
 		b	97f
+85:		ldr	\rp, =UART_OFFSET(AM33XX_UART4_BASE)
+		b	98f
 95:		ldr	\rp, =ZOOM_UART_BASE
 		str	\rp, [\tmp, #0]		@ omap_uart_phys
 		ldr	\rp, =ZOOM_UART_VIRT
diff --git a/arch/arm/plat-omap/include/plat/serial.h b/arch/arm/plat-omap/include/plat/serial.h
index 5e141e1..d15ddb2 100644
--- a/arch/arm/plat-omap/include/plat/serial.h
+++ b/arch/arm/plat-omap/include/plat/serial.h
@@ -102,6 +102,7 @@
 #define TI81XXUART2		82
 #define TI81XXUART3		83
 #define AM33XXUART1		84
+#define AM33XXUART4		85
 #define ZOOM_UART		95		/* Only on zoom2/3 */
 
 /* This is only used by 8250.c for omap1510 */
diff --git a/arch/arm/plat-omap/include/plat/uncompress.h b/arch/arm/plat-omap/include/plat/uncompress.h
index d5786bd..8686776 100644
--- a/arch/arm/plat-omap/include/plat/uncompress.h
+++ b/arch/arm/plat-omap/include/plat/uncompress.h
@@ -190,6 +190,9 @@ static inline void __arch_decomp_setup(unsigned long arch_id)
 
 		/* AM33XX base boards using UART1 */
 		DEBUG_LL_AM33XX(1, am335xevm);
+
+		/* AM33XX IA boards using UART4 */
+		DEBUG_LL_AM33XX(4, am335xiaevm);
 	} while (0);
 }
 
-- 
1.7.11.2

